core/src/main/java/tc/oc/pgm/consumable/ConsumableModule.java

```diff
@@ -0,0 +1,88 @@
+package tc.oc.pgm.consumable;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.logging.Logger;
+import org.jdom2.Document;
+import org.jdom2.Element;
+import org.jetbrains.annotations.Nullable;
+import tc.oc.pgm.action.Action;
+import tc.oc.pgm.action.ActionModule;
+import tc.oc.pgm.action.ActionParser;
+import tc.oc.pgm.api.map.MapModule;
+import tc.oc.pgm.api.map.factory.MapFactory;
+import tc.oc.pgm.api.map.factory.MapModuleFactory;
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.api.module.exception.ModuleLoadException;
+import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.pgm.kits.KitNode;
+import tc.oc.pgm.util.xml.InvalidXMLException;
+import tc.oc.pgm.util.xml.Node;
+import tc.oc.pgm.util.xml.XMLUtils;
+
+public class ConsumableModule implements MapModule<ConsumableMatchModule> {
+
+  private final ImmutableSet<ConsumableDefinition> consumableDefinitions;
+
+  private ConsumableModule(ImmutableSet<ConsumableDefinition> consumableDefinitions) {
+    this.consumableDefinitions = consumableDefinitions;
+  }
+
+  @Override
+  public @Nullable ConsumableMatchModule createMatchModule(Match match) throws ModuleLoadException {
+    return new ConsumableMatchModule(match, consumableDefinitions);
+  }
+
+  public static class Factory implements MapModuleFactory<ConsumableModule> {
+
+    @Override
+    public Collection<Class<? extends MapModule<?>>> getWeakDependencies() {
+      return ImmutableList.of(ActionModule.class);
+    }
+
+    @Override
+    public @Nullable ConsumableModule parse(MapFactory factory, Logger logger, Document doc)
+        throws InvalidXMLException {
+      ActionParser actionParser = new ActionParser(factory);
+
+      Set<ConsumableDefinition> consumableDefinitions = new HashSet<>();
+
+      List<Element> consumableElements =
+          XMLUtils.flattenElements(doc.getRootElement(), "consumables", "consumable");
+      if (consumableElements.size() == 0) return null;
+
+      for (Element consumableElement : consumableElements) {
+        String id = XMLUtils.getRequiredAttribute(consumableElement, "id").getValue();
+        boolean preventDefault =
+            XMLUtils.parseBoolean(consumableElement.getAttribute("prevent-default"), true);
+
+        Action<? super MatchPlayer> action;
+        Node actionNode = Node.fromAttr(consumableElement, "action", "kit");
+        if (actionNode != null) {
```

Comment:
I wonder, does it even make sense to have a consumable with no action? with the functionality offered in this PR, i feel like this shouldn't be optional as simply preventing default and doing nothing seems kinda odd.
Maybe mapdevs can give more input on this?

If it was to be removed, you can simply change to `Node.fromRequiredAttr` and remove the if